secondarylib-feedback/src/main/scala/com/socrata/datacoordinator/secondary/feedback/FeedbackContext.scala

```diff
@@ -116,18 +116,22 @@ class FeedbackContext[CT,CV](user: String,
                              rows: IndexedSeq[Row[CV]],
                              targetColumns: Set[UserColumnId]): Either[ComputationFailure, Map[Int, Map[UserColumnId, CV]]] = {
     val perDatasetData = computationHandler.setupDataset(cookie)
-    val perColumnData = cookie.strategyMap.toSeq.collect {
+    val perColumnDataTargetColMap = cookie.strategyMap.toSeq.collect {
       case (targetCol, strat) if targetColumns.contains(targetCol) & computationHandler.matchesStrategyType(strat.strategyType) =>
-        computationHandler.setupColumn(perDatasetData, strat, targetCol)
-    }
+        (computationHandler.setupColumn(perDatasetData, strat, targetCol) -> targetCol)
+    }.toMap
+    val perColumnData = perColumnDataTargetColMap.keys.toSeq
     val toCompute = rows.iterator.zipWithIndex.flatMap { case (row, index) =>
       val rcis =
         perColumnData.flatMap { columnData =>
           // don't compute if there has been to change to the source columns
-          if (noChange(row, columnData.strategy.sourceColumnIds))
+          val targetColId = perColumnDataTargetColMap(columnData)
+          // New value of either target or source will have the same existing if update does not contain the field.
+          // There should be no change either they contain the same value or are not passed in
+          if (noChange(row, columnData.strategy.sourceColumnIds :+ targetColId))
             None
           else
-            Some(computationHandler.setupCell(columnData, row.data))
+            Some(computationHandler.setupCell(columnData, row))
```

Comment:
Why push this `.data` dereference down into `setupCell`?  Also it makes the imports rather confusing, since down below in the other two classes you have both `com.socrata.datacoordinator.secondary.feedback.Row` and `com.socrata.datacoordinator.secondary.Row` in scope unqualified (the latter via wildcard-import, which grr someone's IDE was overzealous about collapsing multiple imports into a wildcard import, that is my least favorite intellij misfeature) and it confused me about how this change could even compile for a bit.

Reply:
Determine when and when not to geocode is still problematic.  Currently, we have a bug where upserting the same value will erase the computed value.  Here is the path:
core upsert (unfortunately set location/point to null).  DC clears the point.  geocoding secondary doesn't even have a chance to re-compute because it detects no change for it only compares source columns excluding target column.  But if we pass in only one copy of row/data to the handler, there are times it will just accept the new target value (which is undesirable) when the first problem is fixed.

Reply:
I actually have a different version that rename feedback.Row to something else.  What if we use FeedbackRow or RowWithPreviousImage?  I was just trying to minimize superficial code change.